src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -331,7 +331,7 @@ final class RiskProvider: RiskProviding {
 
 	private func previousRiskIfExistingAndNotExpired(userInitiated: Bool) -> Risk? {
 		let enoughTimeHasPassed = riskProvidingConfiguration.shouldPerformExposureDetection(
-			lastExposureDetectionDate: store.enfRiskCalculationResult?.calculationDate
+			lastExposureDetectionDate: store.exposureDetectionDate
```

Comment:
@ArturFriesen 
> The check for having a min. time distance between detection calls is already implemented in RiskProvider. The implementation also depends on a timeframe which is configurable over the AppConfiguration.
I propose to solve the problem at this place. I think we should consider to compare against a date, which we set on detection call instead of the calculation date. Because the calculation date is only set if the ENF returns a success and the calculation was successful.

Do I understand your proposal correctly that we should not only check but also setup the timestamp here in this function? E.g. [before this return?](https://github.com/corona-warn-app/cwa-app-ios/blob/34dc7d3c5acea27dd74829ce0d4199f9e38eb032/src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift#L357)
**Pro**: big improvement with regard to locality of the change
**Con**: currently I see a few exit paths after this check but before the call to ENF, which would unnecessarily extend the time for next detection although ENF was not called ([here](https://github.com/corona-warn-app/cwa-app-ios/blob/34dc7d3c5acea27dd74829ce0d4199f9e38eb032/src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift#L368), [here](https://github.com/corona-warn-app/cwa-app-ios/blob/34dc7d3c5acea27dd74829ce0d4199f9e38eb032/src/xcode/ENA/ENA/Source/Services/Exposure%20Transaction/ExposureDetection.swift#L77), [here](https://github.com/corona-warn-app/cwa-app-ios/blob/34dc7d3c5acea27dd74829ce0d4199f9e38eb032/src/xcode/ENA/ENA/Source/Services/Exposure%20Transaction/ExposureDetection.swift#L83), [here](https://github.com/corona-warn-app/cwa-app-ios/blob/34dc7d3c5acea27dd74829ce0d4199f9e38eb032/src/xcode/ENA/ENA/Source/Services/Exposure%20Transaction/ExposureDetection.swift#L47), and [here](https://github.com/corona-warn-app/cwa-app-ios/blob/34dc7d3c5acea27dd74829ce0d4199f9e38eb032/src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift#L304)). As far as I see these execution paths are currently related to reasons like duplicate risk task execution or wrong device time, so maybe we can just ignore that?